shapely/coords.py

```diff
@@ -37,121 +37,67 @@ class CoordinateSequence(object):
     _ndim = None
     __p__ = None
 
-    def __init__(self, parent):
-        self.__p__ = parent
-
-    def _update(self):
-        self._ndim = self.__p__._ndim
-        self._cseq = lgeos.GEOSGeom_getCoordSeq(self.__p__._geom)
+    def __init__(self, coords):
+        self._coords = coords
```

Comment:
What exactly to do with this `CoordinateSequence` class is something still to be discussed, and I opened https://github.com/Toblerity/Shapely/issues/984 for that

Reply:
I have been reading up on https://github.com/Toblerity/Shapely/issues/984 and I tend to go with this implementation for this PR (making `self._coords` a ndarray directly)

However I think that sticking a full duplicate of `coords` to the object by default is a bit too much. Also `_coords` won't be initialized properly in all construction paths.

Within this approach (I actually like the lazy CoordinateSequency object approach better) I think it is best to use a pattern like this:

```
@property
def coords(self):
    try:
        coords = getattr(self, "_coords")
    except AttributeError:
        coords = pygeos.get_coordinates(self)
        coords.flags.writeable = False
        self._coords = coords
    return coords
```

Reply:
> However I think that sticking a full duplicate of coords to the object by default is a bit too much. 

Note that this CoordinateSequence is not constructed by default when creating a geometry object. It will only be created once a user accesses the `geom.coords` attribute. 

Now, with the current code, each time the user does `geom.coords`, it will create a new CoordinateSequence object containing a new array, so making this "cached" is certainly a good idea (as well as making the array not writable)